### PR TITLE
update collector core, update awsemfexporter and newrelicexporter

### PR DESCRIPTION
**Pull request by codeboten**
The original intent of this PR was to only update the core, however the awsemfexporter and newrelicexporter both relied on the internal OC format which changed to use attributes instead of labels, and caused the tests to fail.

This change fixes those exporters as well as updates the core.

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
awsemfexporter unit tests fail, not sure if related to the change or it just exposed more flakiness. 
@anuraaga @shaochengwang @mxiamxia can you check as codeowners?

**Comment by tigrannajaryan** [comment]
> awsemfexporter unit tests fail, not sure if related to the change or it just exposed more flakiness.
> @anuraaga @shaochengwang @mxiamxia can you check as codeowners?

@anuraaga @shaochengwang @mxiamxia this is blocking the release, can you please have a look?

**Comment by bogdandrutu** [comment]
This is blocked by unittests

**Comment by sodabrew** [comment]
Are the failures isolated to `awsemfexporter` ? Is there anything a contributor could look at to help resolve the blocker?

**Comment by codeboten** [comment]
Rebased.

**Comment by codeboten** [comment]
Please take a look again, I updated the awsemfexporter and newrelicexporter to use attributes instead of labels to get the tests to pass in this PR. I can break it up into separate PRs but have limited time to do it this week and next.

File: exporter/awsemfexporter/datapoint.go
Comment (anuraaga) [inline exporter/awsemfexporter/datapoint.go:204]: Does StringVal work for non string type attrs too?

File: exporter/awsemfexporter/datapoint_test.go
Comment (anuraaga) [inline exporter/awsemfexporter/datapoint_test.go:337]: Can we add tests with non-string attributes?

**Review by anuraaga — APPROVED** [review]

File: exporter/awsemfexporter/datapoint.go
Comment (codeboten) [reply exporter/awsemfexporter/datapoint.go:204]: StringVal will return an empty string if the type is not a string. There is an issue to track the work to transform attributes of any type to strings https://github.com/open-telemetry/opentelemetry-collector/issues/3815

**Comment by bogdandrutu** [comment]
Already merged :)

File: exporter/awsemfexporter/datapoint.go
Comment (mxiamxia) [reply exporter/awsemfexporter/datapoint.go:204]: I am making the changes for emfexporter @codeboten

**Pull request closed without merging**
